pybossa/exporter/task_csv_export.py

```diff
@@ -148,11 +149,44 @@ def _format_csv_row(self, row, headers):
     def _handle_row(self, writer, t, headers):
         writer.writerow(self._format_csv_row(self.merge_objects(t),
                                              headers=headers))
+
+    def _flatten_json_iterative(self,dictionary):
```

Comment:
You might be able to simplify this a bit:

```
def flatten(key_value_pairs, key_prefix='', return_value=None):
    return_value = return_value if return_value is not None else {}
    for k, v in key_value_pairs:
        key = '{}_{}'.format(prefix, k)
        if isinstance(v, dict):
            flatten(v.iteritems(), key, return_value)
        elif isinstance(v, list):
            flatten(enumerate(v), key, return_value)
        else:
            return_value[key] = return_value
    return return_value
```

But yours might have some different behavior.

Reply:
Looks good, use `six.iteritems`

Reply:
Yes, will do it this way. thanks

Reply:
The iterator is nice.  This should work:

```python
def flatten(key_value_pairs, key_prefix='', return_value=None):
    return_value = return_value if return_value is not None else {}
    for k, v in key_value_pairs:
        key = k if not key_prefix else '{}_{}'.format(key_prefix, k)
        if isinstance(v, dict):
            iterator = flatten(v.items(), key, return_value)
        elif isinstance(v, list):
            iterator = flatten(enumerate(v), key, return_value)
        else:
            iterator = [(key, v)]
        for kk, vv, in iterator:
            yield kk, vv


print(dict(flatten({
    'a': [0, 1],
    'b': {
        'c': 1
    },
    'd': 1
}.items())))
```